de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/dvlm/mat/MatImporterTest.java

```diff
@@ -0,0 +1,343 @@
+		assertEquals(tc1.getTestFloat(), tc.getTestFloat(), 0);
+		assertEquals("same testResource", tc1.getTestResource(), tc.getTestResource());
+		assertEquals("same testEnum", tc1.getTestEnum(), tc.getTestEnum());
+		
+	}
+
+	@Test
+	public void testImportOfValuesAddAll() throws IOException {
+		TestCategoryAllProperty tc = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc);
+		tc.setTestBool(true);
+		tc.setTestFloat(2);
+		tc.setTestEnum("HIGH");
+		tc.setTestString("test");
+		tc.setTestInt(1);
+		URI testUri = URI.createPlatformPluginURI("Testresource", true);
+		tc.setTestResource(testUri);
+		mat = exporter.exportSei(sei);
+
+		TestStructuralElement tsei2 = new TestStructuralElement(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setUuid(sei.getUuid());
+		TestCategoryAllProperty tc1 = new TestCategoryAllProperty(testConcept);
+		tsei2.add(tc1);
+		importer.importSei(sei2, mat);
+
+		EList<APropertyInstance> caSei = sei.getCategoryAssignments().get(0).getPropertyInstances();
+		EList<APropertyInstance> caSei2 = sei2.getCategoryAssignments().get(0).getPropertyInstances();
+		assertTrue("same number of elements", caSei.size() == caSei2.size());
+		assertEquals("same testString", tc1.getTestString(), tc.getTestString());
+		assertEquals("same testBool", tc1.getTestBool(), tc.getTestBool());
+		assertEquals(tc1.getTestFloat(), tc.getTestFloat(), 0);
+		assertEquals("same testResource", tc1.getTestResource(), tc.getTestResource());
+		assertEquals("same testEnum", tc1.getTestEnum(), tc.getTestEnum());
+		assertEquals("same testInt", tc1.getTestInt(), tc.getTestInt());
+	}
+
+	@Test
+	public void testImportOfValuesChangeNothingEmpty() throws IOException {
+		TestCategoryAllProperty tc = new TestCategoryAllProperty(testConcept);
+		tsei.add(tc);
+		mat = exporter.exportSei(sei);
+
+		TestStructuralElement tsei2 = new TestStructuralElement(testConcept);
+		StructuralElementInstance sei2 = tsei2.getStructuralElementInstance();
+		sei2.setName("testsei");
+		sei2.setUuid(sei.getUuid());
+		TestCategoryAllProperty tc1 = new TestCategoryAllProperty(testConcept);
+		tsei2.add(tc1);
+		importer.importSei(sei2, mat);
+
+		EList<APropertyInstance> caSei = sei.getCategoryAssignments().get(0).getPropertyInstances();
+		EList<APropertyInstance> caSei2 = sei2.getCategoryAssignments().get(0).getPropertyInstances();
+		assertTrue("same number of elements", caSei.size() == caSei2.size());
+		assertEquals("same testString", tc1.getTestString(), tc.getTestString());
+		assertEquals("same testBool", tc1.getTestBool(), tc.getTestBool());
+		assertEquals(tc1.getTestFloat(), tc.getTestFloat(), 0);
+		assertEquals("same testResource", tc1.getTestResource(), tc.getTestResource());
+		assertEquals("same testEnum", tc1.getTestEnum(), tc.getTestEnum());
+	}
+
+	@Test
+	public void testImportOfValuesChangeNothingValues() throws IOException {

```

Comment:
ChangeNothingValues? TestCase with no updates?

Reply:
A TestCase with no Values is exported and than imported. The test check if the sei doesn't change by this.